libuavcan/src/transport/uc_can_io.cpp

```diff
@@ -361,73 +317,68 @@ int CanIOManager::send(const CanFrame& frame, MonotonicTime tx_deadline, Monoton
         {
             // Building the list of next pending frames per iface.
             // The driver will give them a scrutinizing look before deciding whether he wants to accept them.
+            // TODO: What does this mean? can we optimize further before passing them to the driver?
```

Comment:
This means that the driver must be supplied with a list of pending frames when calling `select`. This is needed to avoid inner priority inversion in the TX queue. This is explained in the section 4.4.3.3 of the spec https://new.uavcan.org/specification/UAVCAN_Specification_v1.0.pdf